src/gen_tw.erl

```diff
@@ -332,16 +387,28 @@ loop(_LVT, LVTUB, [Event = #event{lvt=ELVT}|T], PastEvents, Module, ModStates=[{
             erlang:throw(Reason)
     end.
 
--spec rollback_loop(virtual_time(), event_list(), past_event_list(), virtual_time(), atom(), module_state_list()) -> no_return().
+%% Blocks and waits indefintely for a resume or stop.  After which, the regular
+%% simulation loop is resumed or halted, accordingly.
+-spec pause_loop(virtual_time(), virtual_time(), event_list(), past_event_list(), atom(), module_state_list()) -> no_return().
+pause_loop(LVT, LVTUB, Events, PastEvents, Module, ModStates = [{_, ModState}|_]) ->
```

Comment:
This creates a race condition, namely, if a pause and another relevant system command are sent in rapid succession then they may already be contained in Events.  

I will create a bug to followup on this issue.